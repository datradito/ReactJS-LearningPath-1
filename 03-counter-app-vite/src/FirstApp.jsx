import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle, name } ) => {

    // console.log(props);

  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      {/*<code>{ JSON.stringify( newMessage ) }</code>*/}
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps ={
    name: 'Jassira Ramos',
    subTitle: 'No hay subtitulo',
    //title: 'No hay titulo'
}
