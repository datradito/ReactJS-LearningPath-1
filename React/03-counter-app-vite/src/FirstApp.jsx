import PropTypes from 'prop-types';
// las importo para definirle el tipo a las properties, previamente descargo la dependencia (npm install prop-types);

export const FirstApp = ({ title, subTitle, name  }) => {
    //console.log(title);
    //trabajamos con el condicional, 
   {/* if (!title) {
        throw new Error ('el title no existe');
    } */}
  return (
    <>
    <h1 data-testid="test-title">{ title }</h1>
    <p>{ subTitle }</p>
    <p>{ subTitle }</p>
    <h2>{ name }</h2>
    </>
  )
}
//PROPTYPES DEFINIDOS en main: escribo los valores requeridos
FirstApp.PropTypes = {
title: PropTypes.string.isRequired,
subtitle: PropTypes.number.isRequired
}

{/**caso de utilizacion de numeros:  <p>{subtitle + 1}</p> y en el main  :subtitle = {1233}
 */}

 //tambien podremos definir las propiedades en un objeto: nota: se colocan al final

FirstApp.defaultProps = {
    name:'daiana herrera',
    subtitle: 'no hay sub titulo',
    // en seccion5 title: 'no hay titulo',
}