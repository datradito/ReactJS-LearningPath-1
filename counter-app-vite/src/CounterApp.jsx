import React from "react";
import { Fragment , useState} from "react";
import PropTypes from 'prop-types';




export const CounterApp = ({value}) => {

    const [counter, setCounter] =  useState(value);

    const  handleAdd = () =>  {
       
        setCounter(counter + 1 )

       /*  if(counter > 10) {
            alert('Legaste a 10!')
        } */
    
    }
    const  handleLess = () =>  {
       
        setCounter(counter - 1 )
    
    }
    const  handleReset= () =>  {
        setCounter(value)
    }

    return(
        <Fragment>
            <div className="container-counter">

                <h2>{counter}</h2>

                <button  aria-label="btn-suma" onClick={ handleAdd}>+</button>
                <button aria-label="btn-resta" onClick={ handleLess}>-</button>
                <button aria-label="btn-reset" onClick={ handleReset}>Reset</button>
            </div>
        </Fragment>
    )

}


CounterApp.propTypes = {

    value:  PropTypes.number.isRequired

}

CounterApp.defaultProps = {
    value:  0
}