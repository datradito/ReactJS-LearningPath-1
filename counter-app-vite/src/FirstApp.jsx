import React from "react";
import { Fragment } from "react";
import PropTypes  from 'prop-types'


const text =  "Welcome to my First App in React";

const person = {
    name: "Paloma",
    age: "21",
    f_color: "purple"
}

const sumar = (a,b) => {
    return a + b;
}

export const FirstApp = ({title , subtitle }) => {


    return(
       <Fragment>
           <header className="haeder">
                <div className="container-nav">

                    <nav className="nav">
                        <ul className="nav-list">
                            <li className="nav-list-item">Home</li>
                            <li className="nav-list-item">About me</li>
                            <li className="nav-list-item">Projects</li>
                        </ul>
                    </nav>

                </div>
                <div className="title" data-testid="test-title">
                    <h1>{title}</h1>
                    
                </div>
                <p>{subtitle}</p>
           </header>
            
           {/*  <h2>Hello World</h2>
            <p>This is my first app in react</p>

            <div class="container">

                <h3>Person</h3>
                <ul>
                    <li>{person.name}</li>
                    <li>{person.age}</li>
                    <li>{person.f_color}</li>
                </ul>

 
                <h3>Person</h3>
                <ul>
                    <li>{ JSON.stringify(person)}</li>
                   
                </ul>
            </div>

            <div>
                <p>La suma da como resultado: <span>{sumar(20,40)}</span> </p>
            </div> 

            <footer>
                <h6>By Paloma Herrera</h6>
            </footer>
        */}
       </Fragment>
    )
}



FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}


FirstApp.defaultProps = {
    title: 'Hola, soy Paloma',
    subtitle:'Mi primera app react'

}