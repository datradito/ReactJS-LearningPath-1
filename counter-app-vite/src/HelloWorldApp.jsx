
/* function App() {

    const name = "Paloma";
    // document.createElement.... 
    return (<h1>Hello World, I am { name }</h1>)

}

export default App; */


export const HelloWorldApp = () => {
    return(
        <div>Hello World!!</div>
    )
}