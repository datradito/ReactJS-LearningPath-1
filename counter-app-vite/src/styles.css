* {
    margin: 0;
    padding: 0;
    box-sizing: 0;
}

html, body {
    background-color: #fff;
    color: #298157;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1rem;

}
.nav{
    background-color: #fff;
    color: #298157;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1rem;
    font-weight: bolder;
    width: 100%;
    height: 3.5rem;
}

.nav-list{
  list-style: none;
  margin: 0 ;
  padding: 0;
  display: flex;
}

.nav-list-item{
    margin: 1rem 1rem;
    padding: 0;
}

.nav-list-item:hover{
    text-decoration: #298157 solid 2px underline;
}


.title{
    margin-top: 4rem;
    text-align: center;
    font-size: 30px;
    color: #298157;
}

p{
    margin-top: 4rem;
    font-size: 1.5rem;
    font-weight: bolder;
    font-family: Arial, Helvetica, sans-serif;
    color: #298157;
    text-align: center;
}

.container-counter{
    text-align: center;
    margin: 1rem;
}

button {
    padding: 15px;
    font-size: 1.3rem;
    background-color: #298157;
    color: #fff;
    border: #298157;
    border-radius: 5px;
    margin: 5px;
}




footer h6 {
    color: #868686;
    font-size: 1rem;
    font-family: Arial, Helvetica, sans-serif;
    text-align: center;
}

