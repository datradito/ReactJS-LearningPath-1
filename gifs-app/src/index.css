body{
  background: none;
  border: none;
  padding: 2em;
  background-color: black;
  color: white;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  text-align: center;
  justify-content: center;
  
}
::placeholder { 
  color: rgb(244, 215, 255);
}
input{
  text-align: left;
  padding-right: 10px;
  outline: none;
}
input:hover, input:active{
  border: inherit;
  outline: none;
}
button, input{
  background: none;
  border: 0;
  color: inherit;
  background-color: rgb(119, 27, 204);
  color: rgb(255, 255, 255);
  font-size: 1.5em;
  padding: 1em;
}
button{
  cursor: pointer;
}
ol{
  list-style:decimal;
  text-align: justify;
}
.search{
  display: inline-block;
  width: 100%;
  /*background-color: rgb(79, 26, 114);*/
}
.GifGridItem{
  align-content: center;
  margin: 0.6em;
  border: 0.2em solid rgb(132, 0, 255);
}
.GifGridItem p {
  text-align: center;
}
.card-grid{
  align-content: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
