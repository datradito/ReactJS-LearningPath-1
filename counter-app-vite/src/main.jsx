//importamos react 
import React from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from "./FirstApp";
import { CounterApp } from './CounterApp';

import './styles.css';




//creamos el root al que le vamos a agregar el texto.
/* ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWorldApp/>
    </React.StrictMode>
) */



ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/*  <FirstApp/>  */}
         <CounterApp/>

    </React.StrictMode>
)