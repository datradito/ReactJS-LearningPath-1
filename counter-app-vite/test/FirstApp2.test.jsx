import {render, screen}  from '@testing-library/react'
import {FirstApp} from '../src/FirstApp'


describe('Prubeas en <FirstApp/>', () => {

    test('debe hacer match con la captura ', () => {

      
       const {container} = render( <FirstApp title = {title} /> );


        expect(container).toMatchSnapshot();

    })
 
    test('debe mostrar el mensaje "Hola, soy Paloma"', () => {

        
       render( <FirstApp title = {title} /> );

       expect(screen.getByText(title)).toBeTruthy()

    });


    test('debe mostar el titulo en un h1' , () => {
        render( <FirstApp title = {title} /> );
        expect(screnn.getByRole('heading', {level: 1}).innerHTML).toContain(title)
    });



    test('debe de mostrar el subtitulo enviado por props', () => {

        render( <FirstApp title = {title} subtitle={subtitle} /> );


        expect(screen.getAllByText(subtitle).length).toBe(2);


    })

    
    
})