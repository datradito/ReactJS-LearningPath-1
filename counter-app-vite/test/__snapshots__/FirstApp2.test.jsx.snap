// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubeas en <FirstApp/> debe hacer match con la captura  1`] = `
<div>
  <header
    class="haeder"
  >
    <div
      class="container-nav"
    >
      <nav
        class="nav"
      >
        <ul
          class="nav-list"
        >
          <li
            class="nav-list-item"
          >
            Home
          </li>
          <li
            class="nav-list-item"
          >
            About me
          </li>
          <li
            class="nav-list-item"
          >
            Projects
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="title"
      data-testid="test-title"
    >
      <h1>
        Hola, soy Paloma
      </h1>
    </div>
    <p>
      Mi primera app react
    </p>
  </header>
</div>
`;
